Analytics/Classes/Internal/SEGSegmentIntegration.m

```diff
@@ -106,26 +106,18 @@ - (id)initWithAnalytics:(SEGAnalytics *)analytics httpClient:(SEGHTTPClient *)ht
             [self trackAttributionData:self.configuration.trackAttributionData];
         }];
 
-        if ([NSThread isMainThread]) {
-            [self setupFlushTimer];
-        } else {
-            dispatch_sync(dispatch_get_main_queue(), ^{
-                [self setupFlushTimer];
-            });
-        }
+        self.flushTimer = [NSTimer timerWithTimeInterval:self.configuration.flushInterval
```

Comment:
do we need to check which thread we are on?

Reply:
No, we manually add the timer in `NSRunLoop.mainRunLoop` so it'll always be scheduled to run on the main thread:

https://github.com/segmentio/analytics-ios/blob/9f0eecc9846ed123f378ca87bca1afad3053b68e/Analytics/Classes/Internal/SEGSegmentIntegration.m#L115-L116

We needed to check before because we used `scheduledTimerWithTimeInterval` instead of `timerWithTimeInterval`

Reply:
We've had some issues around concurrency - particularly as the threads we initialize integrations is not guaranteed.  

 This causes some integrations and customers to rely on implementation details 
 that have caused a few issues in the past (https://github.com/segmentio/analytics-ios/issues/518, https://github.com/segmentio/analytics-ios/pull/714, https://github.com/segmentio/analytics-ios/issues/683) so I'd like to be a bit careful rolling this out. If you're confident in these changes as is  - let's plan to cut this as a beta release to get a bit more testing through this change!

Reply:
Thanks for pointing this out! If customers are relying on this kind of implementation detail I think it’d be better to spend more time looking into the impact. Moving this to #785.